x/ibc/light-clients/solomachine/types/consensus_state.go

```diff
@@ -36,9 +35,9 @@ func (cs ConsensusState) GetRoot() exported.Root {
 
 // GetPubKey unmarshals the public key into a tmcrypto.PubKey type.
 func (cs ConsensusState) GetPubKey() tmcrypto.PubKey {
-	publicKey, err := std.DefaultPublicKeyCodec{}.Decode(cs.PublicKey)
-	if err != nil {
-		panic(err)
+	publicKey, ok := cs.PublicKey.GetCachedValue().(tmcrypto.PubKey)
```

Comment:
I think the UnpackInterfaces function is needed for consensus state and header now since they have an Any now

Reply:
Probably yes. I'll take care of that.